PIL/GifImagePlugin.py

```diff
@@ -533,8 +533,18 @@ def getheader(im, palette=None, info=None):
 
     # Header Block
     # http://www.matthewflickinger.com/lab/whatsinagif/bits_and_bytes.asp
+
+    version = b"87a"
+    for extensionKey in ["transparency", "duration", "loop"]:
+        if info and extensionKey in info:
+            version = b"89a"
+            break
+    else:
+        if im.info.get("version") == "89a":
+            version = b"89a"
```

Comment:
I've just added a specific test. You're welcome to help out if you want to, but I will get it working sooner or later.